fhir-persistence-blob-app/src/main/java/com/ibm/fhir/persistence/blob/app/CreateContainer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+package com.ibm.fhir.persistence.blob.app;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.config.FHIRRequestContext;
+import com.ibm.fhir.exception.FHIRException;
+import com.ibm.fhir.persistence.blob.BlobContainerManager;
+import com.ibm.fhir.persistence.blob.BlobManagedContainer;
+
+/**
+ * Create the container if it doesn't currently exist
+ */
+public class CreateContainer {
+    private static final Logger logger = Logger.getLogger(CreateContainer.class.getName());
+    private final String tenantId;
+    private final String dsId;
+    private final boolean dryRun;
+
+    /**
+     * Public constructor
+     * @param tenantId
+     * @param dsId
+     * @param dryRun
+     */
+    public CreateContainer(String tenantId, String dsId, boolean dryRun) {
+        this.tenantId = tenantId;
+        this.dsId = dsId;
+        this.dryRun = dryRun;
+    }
+
+    /**
+     * Create the container for the configured tenant and datasource pair
+     */
+    public void run() throws FHIRException {
+        // Set up the request context for the configured tenant and datastore
+        FHIRRequestContext.set(new FHIRRequestContext(tenantId, dsId));
```

Comment:
was surprised to see FHIRRequestContext here since I thought this utility was for use outside of the server.  is it just a convenience thing for holding the tenantId and datastoreId pair?

Reply:
Yeah, we need to use it for multi-tenant/datasource support because I'm using the same class to manage connections to the blob storage service - which hides the logic to make sure you use the correct connection.

Reply:
got it